mindmeld/active_learning/plot_manager.py

```diff
@@ -0,0 +1,471 @@
+    """
+
+    def __init__(self, experiment_dir_path: str):
+        """
+        Args:
+            experiment_dir_path (str): Path to the experiment directory
+        """
+        self.experiment_dir_path = experiment_dir_path
+        self.accuracies_data = self.get_accuracies_json_data()
+        self.queries_data = self.get_queries_json_data()
+
+    # Get JSON Data
+    def get_accuracies_json_data(self) -> Dict:
+        """Loads accuracies.json from the experiment directory path.
+        Returns:
+            data (dict): Data loaded from accuracies.json.
+        """
+        accuracies_json_path = AL_ACCURACIES_PATH.format(
+            experiment_folder=self.experiment_dir_path
+        )
+        with open(accuracies_json_path, "r") as infile:
+            return json.load(infile)
+
+    def get_queries_json_data(self) -> Dict:
+        """Loads selected_queries.json from the experiment directory path.
+        selected_queries.json stores the queries selected by active learning
+        at each iteration.
+        Returns:
+            data (dict): Data loaded from selected_queries.json.
+        """
+        selected_queries_json_path = AL_SELECTED_QUERIES_PATH.format(
+            experiment_folder=self.experiment_dir_path
+        )
+        with open(selected_queries_json_path, "r") as infile:
+            return json.load(infile)
+
+    def queries_json_data_has_data(self) -> Dict:
+        """Checks whether queries.json is empty {}.
+        Returns:
+            has_data (bool): Whether queries.json has data.
+        """
+        return {} != self.get_queries_json_data()
+
+    # Plotting Meta Functions
+    def generate_plots(self):
+        """Entry point for generating plots as per the specifications provided in the config."""
+        logger.info("Starting to generate plots")
+        plot_functions = [
+            "plot_single_epoch",
+            "plot_avg_across_epochs",
+            "plot_all_epochs",
+        ]
+        for plot_function in plot_functions:
+            logger.info("Plotting: %s", plot_function)
+            plot_function = getattr(PlotManager, plot_function)
+            self.plotting_wrapper(plot_function)
+        if self.queries_json_data_has_data():
+            logger.info("Plotting: plot_stacked_bar")
+            self.plot_stacked_bar()
+
+    def plotting_wrapper(
+        self,
+        function,
+        plot_domain: bool = True,
+        plot_intents: bool = True,
+        plot_entities: bool = False,
+    ):
+        """Plotting wrapper functions for plots that use data from accuracies.json
+        Args:
+            function (generator): plotting function.
+            plot_domain (bool): Whether to generate plots at the domain level.
+            plot_intents (bool): Whether to generate plots at the intent level.
+            plot_entities (bool): Whether to generate plots at the entity level.
+        """
+        if plot_domain:
```

Comment:
Nit: too many tabs. Use the inverse condition to reduce it. For example:
```
if not plot_domain:
    continue
.
.
.



```

Reply:
Resolved, using the inverse condition to reduce tabs